### DISCUSSION TITLE
Avalonia.iOS app does not start in simulator (The request was denied by service delegate (SBMainWorkspace))

### DISCUSSION THREAD

**Discussion by llfab** [AvaloniaUI/Avalonia#17031] [Q&A] [answered]
I have created a minimal sample app here: https://github.com/llfab/Samples/tree/main/AvaloniaiOSApplication1

I have fixed the basic problem with the simulator by adjusting the runtime identifier and setting the render mode to Metal. Now, the basic error is gone but still the startup fails with:

![image](https://github.com/user-attachments/assets/8a2d6912-b497-40ba-8df9-210ecc0337e1)

Avalonia 11.1.3
MacOS: 14.6.1
Simulator Version: 15.4 (1019.2)
Simulator iPhone 15 with iOS 17.5

Any ideas?

**Comment by kekekeks** [comment, marked as answer]
Are you, by chance, trying to run x64 build on arm host? 
Try running `softwareupdate --install-rosetta`

**Reply by llfab** [reply]
Indeed. Tried arm64 because the Mac mini is arm. But that was even worse. So install Rosetta and then simulator-x64 or -arm64

**Reply by llfab** [reply]
Yes, that did the trick. Which part of Avalonia.iOS makes it necessary to have the x86_64 compatibility layer? Just for understanding...

**Reply by maxkatz6** [reply]
Avalonia.iOS doesn't make this necessary.
iOS simulator 17.2+ have a bug requiring x64 to be used with OpenGL rendering.
But again, it doesn't matter with Metal. And you shouldn't specify any RID at all with iOS builds, if you want to avoid further headache.
